lib/cloudregister/registerutils.py

```diff
@@ -86,23 +88,30 @@ def add_region_server_args_to_URL(api, cfg):
 
 
 # ----------------------------------------------------------------------------
-def clean_hosts_file(domain_name):
+def clean_hosts_file(domain_name, registry_name):
```

Comment:
The registry should have the same domain as the repository server. I don't think we want to support a feature where the repository is configured as 'foo.us` and the registry is setup as `bar.eu`. I think it is fair to expect that the domain name is the same. If someone ever needs these to be separate the we'll have that conversation at that time, for now let's enforce that the registry and repository exist on the same domain.

Reply:
Done

Updated the code to handle domain name only
Updated and added tests to check that behavior